### ISSUE TITLE
Redesigning the Nextflow Website

### ISSUE THREAD

**Issue by evanfloden** [nextflow-io/website#14]

**Comment by tobsecret** [comment]
I like the nextflow.io landing page but the enhancements sound nice. As for the docs, I think updating the FAQ wouldn't hurt. There are some things that come up in the gitter over and over again, that are either not in the documentation ("Why is Nextflow using so much memory and how can I change that? / My Nextflow job gets killed on the cluster, what's going on?" --> use NXF_OPTS to restrict memory usage of Nextflow) or not where people expect them/ are looking for them. If someone would like to collaborate on this and collect issues that are quite commonly asked on gitter, then I'm game.

The Nextflow Scripting section could also use some more detail. There are links out to the Groovy documentation but not all Groovy-isms work perfectly in Nextflow, especially because it's really easy to get confused as a newcomer. For example lots of times errors stem from the fact that users of Nextflow expect a list but are really working on a channel. I know there are the patterns on the nextflow github but I think there is value in having those directly in the scripting section of the docs and particular ones cross-referenced in the FAQ. 

Basically I would have the FAQ be somewhat redundant with the rest of the docs, just to provide another way of searching the docs. What do you think?

**Comment by ewels** [comment]
Great! I think that the website is already pretty good to be honest, but I'm all for better harmonisation of design and resources. On the homepage I think it would help to think about the different reasons that people visit. I can think of a few categories:

* Comparing workflow tools / heard that I should use nextflow and want to see what the overall idea is _(kind of like the first 2 minutes of every nextflow presentation)_
    * I think the site could definitely do a better job of selling its key features a little harder. For example, people love features like execution reports and there's no mention of these ;)
    * Explanation of why people should use nextflow instead of a bunch of bash scripts
    * Perhaps some kind of comparison to other workflow tools? Usage statistics if they exist?
    * Much lower-level explanation of concepts. Homepage kind of assumes that the reader is familiar with containers and a fair amount of technical language.
    * The stuff you mention is good - case studies, examples of users
* People wanting to learn how to get started with nextflow
    * Beginner tutorials, feature list again, best practices, community
* Established users looking to solve a problem, find a feature
    * Full docs, patterns
    * Events, news

A new design could try to triage these different users to the information that they're looking for. I think the top group are probably the people who need a little new content. A gentler intro to software packaging would be nice, and a focus on how nextflow makes what can be a very complicated and overwhelming world of technical jargon super easy.

A few specific ideas that wouldn't be too tricky to implement:

* Better gitter integration with the website? eg. using [gitter sidecar](https://sidecar.gitter.im/)
    * Could also have similar integration of the google groups forum, maybe just with a simple iframe
* Less of a jump from `hello world` to `full pipeline`. The examples are really good - perhaps these could be better highlighted under the "quick start"?
* Harmonisation of patterns and documentation, so that it at all looks like it's all the same website (preferably on the same domain too).
* A search that works across _all content_

One of my take-homes from https://multiqc.info and https://clusterflow.io is that people love short videos and in-page examples that they can play with without leaving the browser. And easter-eggs :)

Phil

**Comment by pditommaso** [comment]
Thanks a lot your feedback. These are very valuable suggestions. Some of them can be implemented quickly e.g. a better FAQ, a comparison with other tools, etc. The remaining will try to implement in the upcoming web site restyling. 

ps. the gitter sidecar is already there! bottom-right corner .. :)

**Comment by tobsecret** [comment]
As for the FAQ, should I open a separate issue for that, so we can collect suggestions there? I'm happy to spear head that.

**Comment by pditommaso** [comment]
Thanks, but it's not needed. I've created a skeleton for the FAQ page as [markdown document](https://github.com/nextflow-io/website/blob/faq/content/faq.md). Contributions are welcomed.
